manifests/application.pp

```diff
@@ -21,31 +21,38 @@
   Boolean $use_pulp_2_for_docker = false,
   Boolean $use_pulp_2_for_yum = false,
   Stdlib::Absolutepath $repo_export_dir = '/var/lib/pulp/katello-export',
+  Stdlib::Absolutepath $pulp_client_ca_cert = '/etc/foreman-pki/certs/ca/ca.crt',
+  Stdlib::Absolutepath $pulp_client_cert = '/etc/foreman-pki/certs/foreman/foreman-to-pulp.crt',
+  Stdlib::Absolutepath $pulp_client_key = '/etc/foreman-pki/certs/foreman/foreman-to-pulp.key',
+  Stdlib::Absolutepath $candlepin_ca_cert = '/etc/foreman-pki/certs/ca/ca.crt',
+  Stdlib::Absolutepath $candlepin_events_ssl_cert = '/etc/foreman-pki/certs/foreman/foreman-to-candlepin.crt',
+  Stdlib::Absolutepath $candlepin_events_ssl_key = '/etc/foreman-pki/certs/foreman/foreman-to-candlepin.key',
+  Stdlib::Absolutepath $crane_ca_cert = '/etc/foreman-pki/certs/ca/ca.crt',
 ) {
   include foreman
-  include certs
-  include certs::apache
-  include certs::candlepin
-  include certs::foreman
-  include certs::pulp_client
   include katello::params
+  include foreman::plugin::tasks
 
-  foreman_config_entry { 'pulp_client_cert':
-    value          => $certs::pulp_client::client_cert,
-    ignore_missing => false,
-    require        => [Class['certs::pulp_client'], Foreman::Rake['db:seed']],
+  file { $candlepin_events_ssl_key:
+    group => $foreman::group,
+    mode  => '0640',
   }
```

Comment:
Can we also reuse the Foreman client cert, at least by default? Perhaps it needs application changes, but it makes the application easier.